firefox-ios/Client/Frontend/Browser/Search/SearchViewModel.swift

```diff
@@ -387,6 +399,16 @@ class SearchViewModel: FeatureFlaggable, LoaderListener {
         }
     }
 
+    // We only care about if the section has shown at least one recent search
+    // so we check if recent searches is empty or not.
+    func recordRecentSearchesDisplayedEvent() {
+        guard !recentSearches.isEmpty && shouldShowRecentSearches else { return }
+        if searchTelemetry.hasSeenRecentSearches == false {
```

Comment:
small nit: same comment as above
```suggestion
        if !searchTelemetry.hasSeenRecentSearches {
```